chart/openfaas/templates/istio-mtls.yaml

```diff
@@ -1,5 +1,23 @@
 {{- $functionNs := default .Release.Namespace .Values.functionNamespace }}
 {{- if .Values.istio.mtls -}}
+apiVersion: security.istio.io/v1beta1
+kind: PeerAuthentication
+metadata:
+  name: default
+  namespace: {{ .Release.Namespace }}
+spec:
+  mtls:
+    mode: STRICT
+---
+apiVersion: security.istio.io/v1beta1
+kind: PeerAuthentication
+metadata:
+  name: default
+  namespace:  {{ $functionNs }}
+spec:
+  mtls:
+    mode: STRICT
+---
 # enforce mTLS to openfaas control plane
 apiVersion: authentication.istio.io/v1alpha1
```

Comment:
This doesn't exist in most Istio installs, can we remove it or put it behind some legacy flag?

Reply:
And the DestinationRules as well. They exist but are bad practice

Reply:
Thanks for looking into this. What would you recommend instead? We're looking for an equivalent mTLS configuration.

Reply:
I would recommend no config at all personally. Istio already automatically adds mTLS (https://istio.io/latest/docs/tasks/security/authentication/authn-policy/#auto-mutual-tls).

These configurations are just about making enforcement strict. If the user wants this, they should configure it themselves with standard Istio APIs instead of through helm values IMO. They can even do it mesh wide, etc so it may not always make sense to do it through helm.

This may just be my bias for not liking including the "kitchen sink" in helm charts though..

Reply:
We are mainly doing that because it was done that way before and most of us are relatively light users of service mesh here, so we're here to learn.

If you think we can do away with these settings and have a sane default I'm for that.

Reply:
I think its reasonable to have no options. This matches the linkerd install (no config, but use mtls where possible), and is similar to [knative](https://knative.dev/development/install/installing-istio/#using-istio-mtls-feature) (they don't create a STRICT mode config in the function namespaces, but they do require explicitly disabling mTLS for some components; for openfaas everything supports mtls so there should be no need here).

Reply:
Sounds like a plan to me. No discernible issues with NATS?

Reply:
Communication to NATS will be plaintext since it has no sidecar. You can also add a sidecar (remove the annotation disabling it) and it would start using mtls automatically

Reply:
That might have been something added because NATS wasn't supported at the time. Users would want "full mTLS" so perhaps we can do that here too?

Reply:
@alexellis and @howardjohn  I have removed the additional istio config objects (policies and desinationrules) and I remove the injection label on nats, so it will not get a sidecar injected automatically. I tested this version of the chart and everything seems to work as expected, including an async invoke, see the new testing notes on the PR description.